tutorials/01_introductory/lifecycle.py

```diff
@@ -0,0 +1,266 @@
+"""
+=======================
+The Lifecycle of a Plot
+=======================
+
+Going from start to finish of one plot in Matplotlib.
+
+This tutorial aims to show the beginning, middle, and end of a single
+visualization using Matplotlib. We'll begin with some raw data, and
+end by saving a figure of a customized visualization. Along the way we'll try
+to highlight some neat features and best-practices using Matplotlib.
+
+.. currentmodule:: matplotlib
+
+.. note::
+
+    This tutorial is based off of
+    `this excellent blog post <http://pbpython.com/effective-matplotlib.html>`_
+    by Chris Moffitt. It was modified and put here by Chris Holdgraf.
+
+A note on Object-oriented vs Pyplot
+===================================
+
+Matplotlib has two interfaces. The first is based on MATLAB and uses
+a state-based interface. This is encapsulated in the :mod:`pyplot`
+module. See the :ref:`pyplot tutorials <sphx_glr_tutorials_01_introductory_lifecycle.py>`
+for a more in-depth look at the pyplot interface.
+
+The second option is an an object-oriented (OO) interface.
+In this case, we utilize an instance of :class:`axes.Axes` in order to
+render visualizations on an instance of :class:`figure.Figure`.
+We call methods that do the plotting directly from the
+Axes object, which gives us much more flexibility and
+power over customizing our plot. See the
+:ref:`object-oriented examples <api_examples>` for many examples of how
+this approach is used.
+
+.. note::
+
+    In general, try to use the object-oriented interface over the pyplot
+    interface.
+
+Our data
+========
+
+We'll generate some fake data that is broken down according to groups.
+These data contain one number per group, perhaps representing the mean value
+of each group.
+
+"""
+
+# sphinx_gallery_thumbnail_number = 10
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.ticker import FuncFormatter
+from string import ascii_lowercase
+
```

Comment:
I think tutorials are easier to follow when the sample data is concrete. Here's one example:

```
data = {'apples':10, 'oranges':15, 'lemons':5, 'limes':20}
```